src/ie9-ajax.js

```diff
@@ -58,8 +58,17 @@ function ie9Ajax(method, url, opts) {
     xdr.onerror = handleError;
     xdr.ontimeout = handleError;
 
-    xdr.open(method, authToUrl(url, opts));
-    xdr.send(opts.data);
+    try {
+      xdr.open(method, authToUrl(url, opts));
+      xdr.send(opts.data);
+    } catch (error) {
+      // using trim because the error message has a bunch of white space for some reason
+      if (error.message.trim() === 'Access is denied.') {
+        reject(new Error('Please ensure Shopify JavaScript Buy SDK is run over https'));
```

Comment:
Should this be handled by the logger also? I'm assuming not because maybe there's a case where the user could recover from it.
https://github.com/Shopify/js-buy-sdk/blob/81c4086abd1da4fa707feaba9594ab26d12a2dbb/src/ie9-ajax.js#L55